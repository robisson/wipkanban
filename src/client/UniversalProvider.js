import React from 'react';
import {Provider} from 'react-redux';
import RouterFactory from './routes/RouterFactory'
import configureStore from './configureStore'

const UniversalProvider = ({
    server = false,
    location,
    store
}) => {

    return (
        <Provider store={store}>
            <RouterFactory location={location} server={server}/>
        </Provider>
    )
};

export default UniversalProvider;