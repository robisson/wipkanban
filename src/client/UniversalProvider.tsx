import * as React from "react";
import { Provider } from "react-redux";
import RouterFactory from "./routes/RouterFactory";

interface IUniversalProvider {
  server: boolean;
  location: string;
  store: any;
}

const UniversalProvider: React.StatelessComponent<any> = ({
  server = false,
  location,
  store
}: IUniversalProvider) => {
  return (
    <Provider store={store}>
      <RouterFactory location={location} server={server} />
    </Provider>
  );
};

export default UniversalProvider;
