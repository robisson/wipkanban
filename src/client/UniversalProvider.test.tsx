import * as React from "react";
import UniversalProvider from "./UniversalProvider";
import {shallow} from "enzyme";
import { Provider } from "react-redux";

describe("<UniversalProvider /> Component", () => {
  let wrapper;

  it("should to render a Provider component", () => {
    wrapper = shallow(<UniversalProvider />);
    expect(wrapper.type()).toEqual(Provider);
    expect(wrapper).toMatchSnapshot();
  });
});
