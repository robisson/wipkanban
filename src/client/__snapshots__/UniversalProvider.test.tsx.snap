// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UniversalProvider /> Component should to render a Provider component 1`] = `
<Provider>
  <RouterFactory
    server={false}
  />
</Provider>
`;
